File: springbootDynamoDB-sdk-v2/src/main/java/com/dynamodb/dto/Song.java
package com.dynamodb.dto;

import java.util.Map;

import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
import software.amazon.awssdk.services.dynamodb.model.AttributeValue;

@DynamoDbBean
public class Song {
	
	private String artistName;
	private String songName;
	private String albumTitle;
	
	public Song() {}
	
	public Song(String artistName, String songName, String albumTitle) {
		this.artistName = artistName;
		this.songName = songName;
		this.albumTitle = albumTitle;
	}

	public String getArtistName() {
		return artistName;
	}
	
	public void setArtistName(String artistName) {
		this.artistName = artistName;
	}

	public String getSongName() {
		return songName;
	}
	
	public void setSongName(String songName) {
		this.songName = songName;
	}

	public String getAlbumTitle() {
		return albumTitle;
	}
	
	public void setAlbumTitle(String albumTitle) {
		this.albumTitle = albumTitle;
	}

	public static Song attributeMapToSong(Map<String, AttributeValue> attributeMap) {
		String artistName = attributeMap.get("ArtistName").s();
		String songName = attributeMap.get("SongName").s();
		String albumTitle = attributeMap.get("AlbumTitle").s();
		return new Song(artistName, songName, albumTitle);
	}

	@Override
	public String toString() {
		return "Song [artistName=" + artistName + ", songName=" + songName + ", albumTitle=" + albumTitle + "]";
	}

}

File: springbootDynamoDB-kotlin-sdk-v2/src/main/kotlin/me/brunosantana/dto/Song.kt
package me.brunosantana.dto

import com.fasterxml.jackson.annotation.JsonInclude
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbAttribute
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean
import software.amazon.awssdk.services.dynamodb.model.AttributeValue

private const val TYPE = "SONG"

@DynamoDbBean
@JsonInclude(JsonInclude.Include.NON_EMPTY)
data class Song (
    @get:DynamoDbAttribute("ArtistName")
    var artistName: String,
    @get:DynamoDbAttribute("SongName")
    var songName: String,
    @get:DynamoDbAttribute("AlbumTitle")
    var albumTitle: String
): DynamoBaseModel(
    pkType = "artist",
    pkId = artistName.lowercase().replace(" ", "_"),
    sk = "song#${artistName.lowercase().replace(" ", "_")}#${songName.lowercase().replace(" ", "_")}",
    gsi1pk = "type#song",
    gsi1sk = "type#song",
) {

    @Deprecated(message = "Intended to be used only by AWS SDK")
    constructor() :
            this(
                artistName = "",
                songName = "",
                albumTitle = ""
            )

    @DynamoDbAttribute("Type")
    fun getType(): String {
        return TYPE
    }

    fun setType(type: String) {
        // Do nothing, this setter is just to make the AWS SDK 2.x happy
    }

    companion object {
        fun attributeMapToSong(attributeMap: Map<String, AttributeValue>): Song {
            val artistName = attributeMap["ArtistName"]!!.s()
            val songName = attributeMap["SongName"]!!.s()
            val albumTitle = attributeMap["AlbumTitle"]!!.s()
            return Song(artistName, songName, albumTitle)
        }
    }
}
